packages/block-editor/src/components/link-control/index.js

```diff
@@ -424,6 +424,7 @@ function LinkControl( {
 									) }
 									onChange={ ( { opensInNewTab } ) => {
 										onChange( {
+											...internalControlValue,
```

Comment:
Fixed the disappearance of attributes such as `url` when `open in new tab` is clicked.

Reply:
That's been fixed https://github.com/WordPress/gutenberg/pull/53949

Reply:
Let's rebase this PR.

Reply:
Rebased now. Looks like it's working right.